File: data/code_pairs/fortran/252.f90
module init_conds_module
    implicit none
    contains

    subroutine init_conds(x, y, z, rho, ux, uy, uz, p)
        double precision, intent(in) :: x, y, z
        double precision, intent(out) :: rho, ux, uy, uz, p

        uz = 0.0d0

        if ((x >= 0.25d0) .and. (x <= 0.75d0) .and. (y >= 0.25d0) .and. (y <= 0.75d0)) then
            p = 2.5d0
            rho = 4.0d0
        else
            p = 2.5d0
            rho = 1.0d0
        endif

        ux = 0.5d0
        uy = 0.5d0
    end subroutine init_conds

end module init_conds_module

File: data/code_pairs/cpp/252.cpp
#ifndef INIT_CONDS_H
#define INIT_CONDS_H

void init_conds(double x, double y, double z, double& rho, double& ux, double& uy, double& uz, double& p) {
    uz = 0.0;

    if (x >= 0.25 && x <= 0.75 && y >= 0.25 && y <= 0.75) {
        p = 2.5;
        rho = 4.0;
    } else {
        p = 2.5;
        rho = 1.0;
    }

    ux = 0.5;
    uy = 0.5;
}

#endif // INIT_CONDS_H